File: _CodeTopics/LeetCode_contest/weekly/weekly2022/281-[LeetCode周赛宕机]/281_1.py
class Solution(object):
    def countEven(self, num):
        """
        :type num: int
        :rtype: int
        """
        
        res = 0
        for i in range(2, num+1):
            lis = map(int, list(str(i)))
            if sum(lis) & 1 == 0:
                res += 1
        return res
    
"""
https://leetcode-cn.com/submissions/detail/270589104/

71 / 71 个通过测试用例
状态：通过
执行用时: 48 ms
内存消耗: 12.9 MB
"""

File: _CodeTopics/LeetCode/2001-2200/002180--weekly281_1/tran/281_1.cpp
class Solution {
public:
    int countEven(int num) {
        
        // 第 281 场周赛第一题
        // https://leetcode.cn/submissions/detail/270589104/

        int res = 0;
        for (int i = 2; i < num + 1; ++i) {
            int sum = 0;
            string s = to_string(i);
            for (auto ch : s) {
                sum += ch - '0';
            }
            if (sum % 2 == 0) {
                res += 1;
            }
        }
        return res;
    }
};

/*
https://leetcode.cn/submissions/detail/393273765/

执行用时：
12 ms
, 在所有 C++ 提交中击败了
0.50%
的用户
内存消耗：
5.8 MB
, 在所有 C++ 提交中击败了
56.44%
的用户
通过测试用例：
71 / 71
*/
